lib/tasks/preflight.ex

```diff
@@ -18,4 +23,17 @@ defmodule Mix.Tasks.Preflight do
       System.halt(:abort)
     end
   end
+
+  def check_hex_version do
+    required_version_num = "0.16.0"
+
+    hex_info = "mix hex.info | grep Hex" |> String.to_char_list |> :os.cmd |> List.to_string
```

Comment:
We can leverage `System.cmd/2` here to avoid the char-list transforms, and `String.contains?/2` to avoid the regex stuff

```elixir
{ hex_version_output, _ } = System.cmd("mix", ["hex", "--version"])
String.contains?(hex_version_output, required_version_num)
```